addons/mail/models/discuss/res_partner.py

```diff
@@ -82,11 +82,11 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
             [('user_ids.active', '=', True)],
             [('partner_share', '=', False)]
         ])
-        if channel.group_public_id.id:
+        if channel_group := (channel.parent_channel_id or channel).group_public_id:
```

Comment:
`channel_group` is confusing as we do have this kind of channel. `allowed_group` maybe?
Or just `group`. The one in the iterable can be just `g` or it will be removed when applied the other PR on top.